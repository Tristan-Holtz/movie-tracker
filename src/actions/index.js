export const addMovies = movies => ({
  type: 'ADD_MOVIES',
  movies
});

export const pageUp = page => ({
  type: 'PAGE_UP',
  page
});

export const pageDown = page => ({
  type: 'PAGE_DOWN',
  page
});

export const addToWatchList = toWatch => ({
  type: 'ADD_FAVORITE',
  toWatch
})
