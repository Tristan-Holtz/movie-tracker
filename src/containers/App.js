import React, { Component } from 'react';
import { fetchMovies } from '../apiCalls/apiCalls'
import './App.css';

export class App extends Component {
  constructor() {
    super()
    this.state = {
      movies: [],
      counter: 1,
      loading: true
    }
  }

  async componentDidMount () {
    const movies = await fetchMovies(this.state.counter)
    this.setState({ movies: movies.results })
  }

  async getMovies() {
    this.pageCounter()
    const movies = await fetchMovies(this.state.counter)
    this.setState({ movies: movies.results })
  }

  pageCounter() {
    this.setState({ counter: this.state.counter + 1 })
  }

  render() {
    return(
      <main>
        <h1>Hello World</h1>
        <button onClick={() => this.getMovies()}>{this.state.counter}</button>
      </main>
    )
  }
}

export default App;
