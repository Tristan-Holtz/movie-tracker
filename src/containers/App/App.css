.app {
  display: flex;
  padding: 0 30px 30px 30px;
  background-color: #0f1124;
}