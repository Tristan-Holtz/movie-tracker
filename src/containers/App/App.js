import React, { Component } from 'react';
import { fetchMovies } from '../../apiCalls/apiCalls';
import { connect } from 'react-redux';
import { addMovies } from '../../actions/index'
import { Route } from 'react-router-dom';
import { Header } from '../../components/header/header'
import MovieContainer from '../MovieContainer/MovieContainer';
import './App.css';

export class App extends Component {
  async componentDidMount () {
    const movies = await fetchMovies(this.props.page)
    this.props.addMovies(movies)
  }

  async getMovies() {
    const movies = await fetchMovies(this.props.page)
    this.props.addMovies(movies)
  }

  render() {
    return(
      <>
        <Header />
        <Route 
          path='/'
          render={() => <MovieContainer />} 
        />
        <Route 
          path='movies/:page' 
          render={() => <MovieContainer />} 
        />
      </>
    )
  }
}

export const mapStateToProps = state => ({
  page: state.page
})

export const mapDispatchToProps = dispatch => ({
  addMovies: movies => dispatch( addMovies(movies) )
})

export default connect(null, mapDispatchToProps)(App);
