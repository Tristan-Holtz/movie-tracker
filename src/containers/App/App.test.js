import React from 'react';
import { shallow } from 'enzyme';
import { addMovies } from '../../actions/index'
import { App, mapStateToProps, mapDispatchToProps } from './App';

it('should match a snapshot', () => {
  const wrapper = shallow(<App />)

  expect(wrapper).toMatchSnapshot();
})

describe('mapStateToProps', () => {
  it('should return an object with a page number and an array of movies', () => {
    const wrapper = shallow(<App />);

    const mockState = {
      page: 1,
      movies: [
        {
            "genre_ids": [
                27,
                9648,
                878,
                53
            ],
            "title": "The Invisible Man",
            "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
            "release_date": "2020-02-26"
        },
        {
            "genre_ids": [
                27,
                9648,
                53
            ],
            "title": "Brahms: The Boy II",
            "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
            "release_date": "2020-02-20"
        }
      ]
    }

    const expected = {
      page: 1,
      movies: [
        {
            "genre_ids": [
                27,
                9648,
                878,
                53
            ],
            "title": "The Invisible Man",
            "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
            "release_date": "2020-02-26"
        },
        {
            "genre_ids": [
                27,
                9648,
                53
            ],
            "title": "Brahms: The Boy II",
            "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
            "release_date": "2020-02-20"
        }
      ]
    }

    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  })
})

describe('mapDispatchToProps', () => {
  it('should call dispatch with an addMovies action when addMovies is called', () => {
    const mockDispatch = jest.fn();
    const movies = [
      {
          "genre_ids": [
              27,
              9648,
              878,
              53
          ],
          "title": "The Invisible Man",
          "overview": "When Cecilia's abusive ex takes his own life and leaves her his fortune, she suspects his death was a hoax. As a series of coincidences turn lethal, Cecilia works to prove that she is being hunted by someone nobody can see.",
          "release_date": "2020-02-26"
      },
      {
          "genre_ids": [
              27,
              9648,
              53
          ],
          "title": "Brahms: The Boy II",
          "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
          "release_date": "2020-02-20"
      }
    ]
    const action = addMovies(movies)
    const mappedProps = mapDispatchToProps(mockDispatch)

    mappedProps.addMovies(movies);
    expect(mockDispatch).toHaveBeenCalledWith(action);
  })
})

