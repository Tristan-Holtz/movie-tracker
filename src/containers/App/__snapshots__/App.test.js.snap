// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
<Fragment>
  <Header />
  <Connect(Search) />
  <NavLink
    to="/watchlist"
  >
    Watchlist
  </NavLink>
  <Switch>
    <Route
      path="/watchlist"
      render={[Function]}
    />
    <Route
      path="/movies/:page"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movie/:id"
      render={[Function]}
    />
    <Route
      path="/"
      render={[Function]}
    />
  </Switch>
</Fragment>
`;
